tinygrad/renderer/triton.py

```diff
@@ -58,14 +58,14 @@ def kk(s): kernel.append("  "*depth+s)
     UnaryOps.SQRT: lambda x,: f"tl.sqrt({x})",
     UnaryOps.NEG: lambda x,: f"-{x}",
     BinaryOps.ADD: lambda x,y,: f"({x}+{y})", BinaryOps.SUB: lambda x,y,: f"({x}-{y})",
-    BinaryOps.MUL: lambda x,y,: f"({x}*{y})", BinaryOps.DIV: lambda x,y,: f"({x}/{y})",
+    BinaryOps.MUL: lambda x,y,: f"({x}*{y})", BinaryOps.DIV: lambda x,y,: f"({x}/{y})" if y != '0.0' else f"{x}*tl.where({x}==0.0, float('nan'), float('inf'))",
     BinaryOps.MAX: lambda x,y,: f"tl.maximum({x},{y})",
     BinaryOps.CMPLT: lambda x,y,: f"({x}<{y})",
-    BinaryOps.MOD: lambda x,y,: f"({x}%{y})",
+    BinaryOps.MOD: lambda x,y,: f"tl.abs({x})%tl.abs({y})*tl.where({x}<0,-1,1)",
```

Comment:
Why does this have to change?

Reply:
Triton uses the same mod as python and it's different than what tinygrad expects eg. `1.0%-3.0==-1` where we expect `2`, its basically copied from https://github.com/tinygrad/tinygrad/blob/36d4001b4f6541724071e2c78cb93e235877902e/test/test_uops.py#L92C141-L92C141. My guess is that when we changed from tl.where to constexpr it moved to doing this in python compile time vs doing it in runtime and it changed the value

Reply:
Ohh..if these are consts. Yea, consts uops should just be constant folded and not passed to here.